### PR TITLE
Add basic Tor controller

### PR DESCRIPTION
**Pull request by bassosimone**

### CODE REVIEW

**Comment by bassosimone** [comment]
@0xPoly can you review this?

File: include/measurement_kit/tor/controller.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+namespace measurement_kit {
+namespace tor {
+
+/// Tor controller class
+class Controller {
+  public:
+    /// Constructor with custom address and port
+    Controller(std::string socks_host, int socks_port,
+               std::function<void()> start_tor)
+        : host_(socks_host), port_(socks_port), start_tor_(start_tor) {}
```
Comment (0xPoly) [inline include/measurement_kit/tor/controller.hpp]: What will happen if tor is started on a port that is too low? (<1024)

**Comment by 0xPoly** [comment]
What happens if internet connection is lost for a long period of time? Will you be able to restart tor when internet is back or will measurement kit have to be restarted?

**Comment by 0xPoly** [comment]
possible enhancement: ability to ask tor for a fresh circuit/new identity

**Comment by bassosimone** [comment]
Thanks! I'm going to merge this as is, but I've added you comments in #183.

**Pull request merged**